client/core/trade.go

```diff
@@ -213,19 +243,43 @@ func (t *trackedTrade) lockRedemptionFraction(num, denom uint64) {
 	if !is {
 		return
 	}
-	newReserved := applyFraction(num, denom, t.redemptionReserves)
-	if t.redemptionLocked+newReserved > t.redemptionReserves {
+
+	newReserved := t.reservesToLock(num, denom, t.redemptionReserves, t.redemptionLocked)
+
+	if err := redeemer.ReReserveRedemption(newReserved); err != nil {
+		t.dc.log.Errorf("error re-reserving redemption %d %s for order %s: %v",
+			newReserved, t.wallets.toAsset.UnitInfo.AtomicUnit, t.ID(), err)
+		return
+	}
+	t.redemptionLocked += newReserved
+}
```

Comment:
This is a change in behavior, even if the outcome may be identical. Before, if the requested reservers were `> redemptionReserves`, an error was logged and that's it. Now, an error is logged, but we still call `ReReserveRedemption` with an argument of `0`, and add `0` to the `redemptionLocked`. I don't expect that this would result in any issues, but calling `ReReserveRedemption` with `0` is questionable behavior.